app/src/main/java/com/r4/notifications/mirror/NotificationGenerator.java

```diff
@@ -0,0 +1,75 @@
+package com.r4.notifications.mirror;
+
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.content.Context;
+import android.os.Build;
+import android.util.Log;
+
+import androidx.core.app.NotificationCompat;
+import androidx.core.app.NotificationManagerCompat;
+
+import static androidx.core.content.ContextCompat.getSystemService;
+
+public class NotificationGenerator {
+    private static final String TAG = "nm.NotificationGenerator";
+
+    static int GENERATOR_ID = 0;
+
+    NotificationCompat.Builder builder;
+    NotificationManagerCompat notificationManager;
+    String CHANNEL_ID;
+    Context context;
+    boolean initialized = false;
+
+
+    public NotificationGenerator(Context context, String textTitle, String textContent) {
+        CHANNEL_ID = "TestChannel_" + GENERATOR_ID++;
+        this.context = context;
+
+        //create the notification channel
+        createNotificationChannel(CHANNEL_ID, CHANNEL_ID);
+
+        //create the notification
+        builder = new NotificationCompat.Builder(context, CHANNEL_ID)
+                .setSmallIcon(android.R.drawable.star_big_on)
+                .setContentTitle(textTitle)
+                .setContentText(textContent)
+                .setPriority(NotificationCompat.PRIORITY_DEFAULT);
+
+        //initialize the notificationManager
+        notificationManager = NotificationManagerCompat.from(context);
+
+        initialized = true;
+    }
+
+
+    private void createNotificationChannel(String channelName, String channelDescription) {
```

Comment:
A channel will only be created upon calling the constructor of the NotificationGenerator, not when a notification is created.
Since only one instance is created at the start of the MainActivity only one channel will be created independet of the amount of test messgages.